### PR TITLE
Added functional tests for node routes in the 2.0 api

### PR DESCRIPTION
**Pull request by geoff-reid**
@abildw @brianparry @srinia6 @dalebremner @RackHD/corecommitters

### CODE REVIEW

File: test/tests/api/v2_0/nodes_tests.py [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+
+    @test(groups=['test-node-id-api2'], depends_on_groups=['test-nodes-api2'])
+    def test_node_id(self):
+        """ Testing GET:/api/2.0/nodes/:id """
+        Api().nodes_get_all();
+        nodes = loads(self.__client.last_response.data)
+        LOG.debug(nodes,json=True)
+        codes = []
+        for n in nodes:
+            LOG.info(n)
```
Comment (jlongever) [inline test/tests/api/v2_0/nodes_tests.py]: fyi passing json=True to LOG will format the JSON output.

File: test/tests/api/v2_0/nodes_tests.py [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+        for n in nodes:
+            if n.get('type') == 'compute':
+                id = n.get('id')
+                assert_is_not_none(id)
+                Api().nodes_del_active_workflow_by_id(identifier=id)
+                assert_equal(0, len(self.__client.last_response.data),
+                        message='No active Workflows found for Node {0}'.format(id))
+        assert_raises(rest.ApiException, Api().nodes_del_active_workflow_by_id, 'fooey')
+
+    @test(groups=['node_tags_patch'])#, depends_on_groups=['nodes.api2.discovery.test'])
```
Comment (jlongever) [inline test/tests/api/v2_0/nodes_tests.py]: You probably want a nodes test dependency.   Just to squelch related error noise caused by dependent failures.

File: test/tests/api/v2_0/nodes_tests.py [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+        check = self.check_compute_count()
+        if check and check == self.__discovered:
+            self.__worker.stop()
+            self.__worker = None
+            self.__discovered = 0
+
+    @test(groups=['test-nodes-api2'], depends_on_groups=['nodes.api2.discovery.test'])
+    def test_nodes(self):
+        """ Testing GET:/api/2.0/nodes """
+        Api().nodes_get_all();
```
Comment (dalebremner) [inline test/tests/api/v2_0/nodes_tests.py]: ; not needed?

**Comment by geoff-reid** [comment]
fixed issues listed above.

**Comment by dalebremner** [comment]
:+1:

**Comment by abildw** [comment]
+1

**Comment by brianparry** [comment]
:+1:

**Pull request merged**